Added branching scenarios

Kalaiselvan88 opened the pull request:
Raising this PR for adding more guidelines reg, the branch count based on discussions that happened while implementing the baseline in projects. Please review

hussainweb commented on content/docs/how-we-work/baseline/git-workflow.md:
```diff
@@ -30,3 +30,25 @@ The branches can be deleted immediately and [automatically](https://docs.github.
+
+#### Scenario 1 (1 branch)
+
+This is the most ideal scenario where we have only 1 branch which is either `main` or `master`. Any other branches are temporary feature branches which would be deleted after the PR is closed. In case of any feature that needs to be reverted, raising a revert PR against the merged one would work.The `main` branch is linked to the Development environment where the next code for Production is pushe ...
```
```suggestion
This is the ideal scenario where we have only one branch (usually `main`). Any other branches are temporary feature branches which would be deleted after the PR is closed. In case of any feature that needs to be reverted, raising a revert PR against the merged one would work. The `main` branch is linked to the Development environment where the next code for Production is pushed and QA is done followed by bug fixes. Individual feature environments are created per branch as required.
```

Kalaiselvan88 replied:
Fixed in 8736b9aa7de8b9982aa0870972d77b6f6afe76b4 and removed above line as well

hussainweb commented:
@Kalaiselvan88, in trying to understand the change, I feel that you are trying to come up with a hybrid approach that keeps some elements of CD while allowing us to meet client requirements. I feel that this is leaving us with the _worst_ of both worlds. My suggestion is to change client expectations around how UAT is done and how deployment is done. If that is not possible, then we have to define a process that is not a compromise. Let's talk about it?

Kalaiselvan88 commented on content/docs/how-we-work/baseline/git-workflow.md:
```diff
@@ -30,3 +30,25 @@ The branches can be deleted immediately and [automatically](https://docs.github.
+
+This is the most ideal scenario where we have only 1 branch which is either `main` or `master`. Any other branches are temporary feature branches which would be deleted after the PR is closed. In case of any feature that needs to be reverted, raising a revert PR against the merged one would work.The `main` branch is linked to the Development environment where the next code for Production is pushe ...
+
+This method is effective for projects with few or no feature changes post User Acceptance Testing (UAT).However, we should withhold merging any tickets into the `main` or `master` branch if they are not planned current release. Instead, these tickets should remain on a feature branch until the appropriate tag is created for the planned deployment continuously rebased against `main` or `master`.
```
@hussainweb Let me explain the scenario where this is happening currently.
1. We have a set of features grouped that are planned in the current release.
2. There is more capacity available in the team and they pick up items from the product backlog that are not on hold but are currently being worked on. These are WIP developments and hence in future releases.

Point 2 is done when we have more Continuous development (CD) hours in large sizes like in MSF so that we work on multiple releases at the same time to utilise the CD hours. Above is the use case, when the work is not on hold, currently being developed.

I agree that ideally, we need to have a feature flag so that this scenario does not happen and we continuously merge things to the `main`. We plan to work on a couple of projects to start with feature flags and gradually roll it out.

I have removed the line in 63e7758aba7240ce4a932805380436a7af4de5c2 as anyway we have mentioned about features staying in feature environments in your suggestion above.

Kalaiselvan88 commented:
> @Kalaiselvan88, in trying to understand the change, I feel that you are trying to come up with a hybrid approach that keeps some elements of CD while allowing us to meet client requirements. I feel that this is leaving us with the _worst_ of both worlds. My suggestion is to change client expectations around how UAT is done and how deployment is done. If that is not possible, then we have to define a process that is not a compromise. Let's talk about it?

@hussainweb I have addressed the comments in Scenario 1. Is this about Scenario 2? If so, sure will schedule a call

Kalaiselvan88 commented on content/docs/how-we-work/baseline/git-workflow.md:
```diff
@@ -30,3 +30,25 @@ The branches can be deleted immediately and [automatically](https://docs.github.
+
+#### Scenario 1 (1 branch)
+
+This is the ideal scenario where we have only one branch (usually `main`). Any other branches are temporary feature branches which would be deleted after the PR is closed. The `main` branch is linked to the Development environment where the next code for Production is pushed and QA is done followed by bug fixes. Individual feature environments are created per branch as required.
```
@hussainweb I wanted to point out one constraint we have with Acquia Cloud or platform.sh due to very little or no default support for additional environments other than default DEV, STAGE, and PROD creating a feature environment for each feature becomes an additional cost. Pantheon is good this way as it supports a default number of Multidevs (10 as per https://docs.pantheon.io/guides/multidev). However in Platform.sh it comes at additonal cost.

> Individual feature environments are created per branch as required.

hussainweb commented:
@Kalaiselvan88, I understand the constraint given the number of environments. I am worried that we are making it harder for developers to understand what they need to do with these scenarios. If Pantheon is the answer, let's explore that. I'd prefer something that we can simplify and reduce to one workflow, not multiple scenarios. This is something better discussed on a call. I see the call invite. Let's talk then.
